.github/workflows/main.yml

```diff
@@ -50,3 +80,31 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           PR_NUMBER: ${{ github.event.pull_request.number }}
+  
+  Generate_Docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install pre-commit
+        run: |
+          python -m pip install --upgrade pip
+          pip install pre-commit
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
```

Comment:
Use the sha instead of the version number

Reply:
Still need to use the sha here.

Reply:
updated the sha